packages/ws-server/src/data-source.ts

```diff
@@ -45,23 +46,27 @@ export function createWSServerDataSource() {
       )
     }
     values.set(key, value)
+    cache.delete(key)
 
     const handlers = clientSubscribers.get(key)
     if (!handlers) return
     handlers.forEach((handler) => handler())
   }
 
   async function get(key: string) {
-    let value = values.get(key)
-    if (typeof value === 'function') {
-      value = await value()
-    }
-    if (isReactElement(value)) {
-      throw new Error(
-        'Rise JSX not configured. You must set "jsx" to "react-jsx" and "jsxImportSource" to "@final-ui/react" in your tsconfig.json.'
-      )
+    if (!cache.has(key)) {
```

Comment:
hmm this change is what you mentioned in the voice message. we will have to work on this for the new model system, because its possible that it already caches